pkg/frontend/querymiddleware/querysharding_test.go

```diff
@@ -1865,3 +1865,88 @@ func TestMapEngineError(t *testing.T) {
 		})
 	}
 }
+
+// Note that this test will fail until these are vendored in.
+// https://github.com/prometheus/prometheus/pull/17608 and https://github.com/grafana/mimir/pull/13676
+func TestConflictingCounterResets(t *testing.T) {
```

Comment:
Might be worth adding some explanation here for what this is testing, as it's not immediately obvious.